mod/node-api/handlers/beacon/historical.go

```diff
@@ -62,13 +67,39 @@ func (h *Handler[_, ContextT, _, _]) GetStateFork(c ContextT) (any, error) {
 	if err != nil {
 		return nil, err
 	}
-	fork, err := h.backend.StateForkAtSlot(slot)
+	forkGeneric, err := h.backend.StateForkAtSlot(slot)
 	if err != nil {
 		return nil, err
 	}
+	fork, err := convertToFork(forkGeneric)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a custom struct for JSON marshaling as per BeaconAPIs.
+	customFork := struct {
+		PreviousVersion common.Version `json:"previous_version"`
+		CurrentVersion  common.Version `json:"current_version"`
+		Epoch           string         `json:"epoch"`
+	}{
+		PreviousVersion: fork.GetPreviousVersion(),
+		CurrentVersion:  fork.GetCurrentVersion(),
+		Epoch:           strconv.FormatUint(uint64(fork.GetEpoch()), 10),
+	}
+
 	return beacontypes.ValidatorResponse{
 		ExecutionOptimistic: false, // stubbed
 		Finalized:           false, // stubbed
-		Data:                types.Wrap(fork),
+		Data:                customFork,
 	}, nil
 }
+
+func convertToFork[T any](forkGeneric T) (*consensustypes.Fork, error) {
+	fork, ok := any(forkGeneric).(*consensustypes.Fork)
```

Comment:
this defeats the point of using generics

Reply:
gotcha, changed to interface.